### PR TITLE
fix: get `network` data from DB directly

### PR DESCRIPTION
**Pull request by wa0x6e**
## 🧿 Current issues / What's wrong ?

The `network` query on the hub is using data from space's cache.
This create refresh lag, and dependency on the `space` cache object.

## 💊 Fixes / Solution

Use a SQL query to get networks count data from the database, and return more network properties, allowing queries like:

```graphql
query Networks {
  networks {
    key
    id
    spacesCount
    name
    chainId
    network
    explorer {
      url
      apiUrl
    }
    logo
    testnet
  }
}
```

## 🚧 Changes

- Get networks count data from the database instead of space's cache
- Return more fields from networks.json. Not all fields from networks.json are returned, only the one used by the snapshot UI.

## 🛠️ Tests

- Query some network data from graphql => it should return same results as before, in addition to the new fields

### CODE REVIEW

**Comment by wa0x6e** [comment]
> But then we will miss new networks , where no one is using them yet

Aren’t they already missing from current implementation ?

**Comment by ChaituVR** [comment]
> Aren’t they already missing from current implementation ?

True lol. i thought we were using `networks.json`,

File: src/graphql/operations/networks.ts [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+export default async function () {
+  const query = `
+    SELECT
+      JSON_UNQUOTE(JSON_EXTRACT(settings, '$.network')) as network,
+      COUNT(name) AS spacesCount
+    FROM spaces
+    GROUP BY network
+    ORDER BY spacesCount DESC;
+  `;
+
```
Comment (ChaituVR) [inline src/graphql/operations/networks.ts:13]: Takes around 0.5 sec on prod, should be fine?

**Comment by wa0x6e** [comment]
> Would be nice if we could add it now 😄 so then maybe we can remove networks.json dependency on snapshot UI

Added

**Comment by wa0x6e** [comment]
Not all fields from networks.json are returned, tell me if we need to add more.

**Comment by wa0x6e** [comment]
Pretty fast sql query, averaging 350ms execution time.

We could save the computation by caching the whole thing, and using `SELECT MAX(updated) FROM spaces;` as cache key

**Comment by wa0x6e** [comment]
Need fix to also include networks from inside strategies

**Pull request closed without merging**